From a18b20b5d3eb1bdec3dca63e1f97808112c2bb94 Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Fri, 6 Feb 2015 14:52:51 +0800
Subject: [PATCH 0455/1594] MLK-10225-2 ARM: imx: move clock enable after set
 parent

commit 9b10a0860c5277c4b10033e78dc3c06bdf2013b7 from
git://git.freescale.com/imx/linux-2.6-imx.git

Move clock enable after parent clock set, otherwise parent clock
set failed.

Signed-off-by: Fugang Duan <B38611@freescale.com>
---
 arch/arm/mach-imx/clk-imx7d.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-imx/clk-imx7d.c b/arch/arm/mach-imx/clk-imx7d.c
index eff3cee..814d908 100644
--- a/arch/arm/mach-imx/clk-imx7d.c
+++ b/arch/arm/mach-imx/clk-imx7d.c
@@ -864,9 +864,6 @@ static void __init imx7d_clocks_init(struct device_node *ccm_node)
 	clk_register_clkdev(clks[IMX7D_GPT1_ROOT_CLK], "ipg", "imx-gpt.0");
 	clk_register_clkdev(clks[IMX7D_GPT_3M_CLK], "gpt_3m", "imx-gpt.0");
 
-	for (i = 0; i < IMX7D_END_CLK; i++)
-		clk_prepare_enable(clks[i]);
-
 	/* use old gpt clk setting, gpt1 root clk must be twice as gpt counter freq */
 	imx_clk_set_parent(clks[IMX7D_GPT1_ROOT_SRC], clks[IMX7D_OSC_24M_CLK]);
 
@@ -891,6 +888,9 @@ static void __init imx7d_clocks_init(struct device_node *ccm_node)
 	imx_clk_set_parent(clks[IMX7D_UART6_ROOT_SRC], clks[IMX7D_PLL_ENET_MAIN_100M_CLK]);
 	imx_clk_set_parent(clks[IMX7D_UART7_ROOT_SRC], clks[IMX7D_PLL_ENET_MAIN_100M_CLK]);
 
+	for (i = 0; i < IMX7D_END_CLK; i++)
+		clk_prepare_enable(clks[i]);
+
 	mxc_timer_init_dt(of_find_compatible_node(NULL, NULL, "fsl,imx7d-gpt"));
 }
 CLK_OF_DECLARE(imx7d, "fsl,imx7d-ccm", imx7d_clocks_init);
-- 
1.7.5.4

